packages/block-library/src/cover/index.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Server-side rendering of the `core/categories` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/categories` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the categories list/dropdown markup.
+ */
+function render_block_core_cover( $attributes, $content ) {
+	if( false === $attributes['useFeaturedImage'] ) {
+		return $content;
+	}
+
+	$currentFeaturedImage = get_the_post_thumbnail_url();
+
+	if( false === $currentFeaturedImage ) {
+		return $content;
+	}
+
+	$isImgElement = ! ( $attributes['hasParallax'] || $attributes['isRepeated'] );
+	$isImageBackground = $attributes['backgroundType'] === 'image';
+
+	if( $isImageBackground && ! $isImgElement ) {
+		$content = preg_replace(
+			'/class=\".*?\"/',
+			'${0} style="background-image:url(' . $currentFeaturedImage . ')"',
```

Comment:
```suggestion
			'${0} style="background-image:url(' . esc_url( $currentFeaturedImage ) . ')"',
```

Should we escape here?

Reply:
Why escape the result of an internal function? If the prop is set wrongfully we're probably having bigger problems?

Reply:
Ah wait you're right as this can be set in various ways.